notify/webhook/webhook.go

```diff
@@ -112,8 +113,17 @@ func (n *Notifier) Notify(ctx context.Context, alerts ...*types.Alert) (bool, er
 		url = strings.TrimSpace(string(content))
 	}
 
+	if n.conf.Timeout != nil {
+		postCtx, cancel := context.WithTimeoutCause(ctx, *n.conf.Timeout, fmt.Errorf("configured webhook timeout (%s) reached", *n.conf.Timeout))
+		defer cancel()
+		ctx = postCtx
+	}
+
 	resp, err := notify.PostJSON(ctx, n.client, url, &buf)
 	if err != nil {
+		if errors.Is(err, context.DeadlineExceeded) && ctx.Err() != nil {
```

Comment:
I think you're right, it's redundant, but I'm going to do the check on Err instead, in case the call returns something else.

I also tweaked the error message a bit, because I think actually maintaining the original error might be valuable, in case some extra information was added there.